go-controller/pkg/ovn/master.go

```diff
@@ -996,63 +991,46 @@ func (oc *Controller) syncNodes(nodes []interface{}) {
 
 	for nodeName, chassisName := range chassisMap {
 		if chassisName != "" {
-			_, stderr, err = util.RunOVNSbctl("--if-exist", "chassis-del", chassisName)
+			cmd, err := oc.ovnSBClient.ChassisDel(chassisName)
 			if err != nil {
-				klog.Errorf("Failed to delete chassis with name %s for logical switch %s: stderr: %q, error: %v",
-					chassisName, nodeName, stderr, err)
+				klog.Errorf("Unable to create the ChassisDel command for chassis: %s from the sbdb", chassisName)
+			} else if err = oc.ovnSBClient.Execute(cmd); err != nil {
+				klog.Errorf("Failed to delete chassis with name %s for node %s: error: %v",
+					chassisName, nodeName, err)
 			}
 		}
 	}
 }
 
-func (oc *Controller) unmarshalChassisDataIntoMap(chData []byte) (map[string]string, error) {
-	//map of node name to chassis name
-	chassisMap := make(map[string]string)
-
-	type chassisList struct {
-		Data     [][]string
-		Headings []string
-	}
-	var mapUnmarshal chassisList
-
-	if len(chData) == 0 {
-		return chassisMap, nil
-	}
-
-	err := json.Unmarshal(chData, &mapUnmarshal)
+func (oc *Controller) deleteNodeChassis(nodeName string) error {
+	var cmds []*goovn.OvnCommand
+	var chNames []string
 
+	chassisList, err := oc.ovnSBClient.ChassisGet(nodeName)
 	if err != nil {
-		return nil, fmt.Errorf("error unmarshaling the chassis data: %s", err)
+		return fmt.Errorf("failed to get chassis list of node %s: error: %v", nodeName, err)
 	}
 
-	for _, chassis := range mapUnmarshal.Data {
-		if len(chassis) < 2 || chassis[0] == "" || chassis[1] == "" {
+	for _, chassis := range chassisList {
+		if chassis.Name == "" {
+			klog.Warningf("Chassis name is empty for node: %s", nodeName)
 			continue
 		}
-		chassisMap[chassis[1]] = chassis[0]
+		cmd, err := oc.ovnSBClient.ChassisDel(chassis.Name)
+		if err != nil {
+			return fmt.Errorf("unable to create the ChassisDel command for chassis: %s", chassis.Name)
+		}
+		chNames = append(chNames, chassis.Name)
+		cmds = append(cmds, cmd)
 	}
 
-	return chassisMap, nil
-}
-
-func (oc *Controller) deleteNodeChassis(nodeName string) error {
-	chassisName, stderr, err := util.RunOVNSbctl("--data=bare", "--no-heading",
-		"--columns=name", "find", "Chassis",
-		"hostname="+nodeName)
-	if err != nil {
-		return fmt.Errorf("failed to get chassis name for node %s: stderr: %q, error: %v",
-			nodeName, stderr, err)
+	if len(cmds) == 0 {
+		return fmt.Errorf("failed to find chassis of node %s", nodeName)
 	}
 
-	if chassisName == "" {
-		klog.Warningf("Chassis name is empty for node: %s", nodeName)
-	} else {
-		_, stderr, err = util.RunOVNSbctl("--if-exist", "chassis-del", chassisName)
-		if err != nil {
-			return fmt.Errorf("failed to delete chassis with name %s for node %s: stderr: %q, error: %v",
-				chassisName, nodeName, stderr, err)
-		}
+	if err = oc.ovnSBClient.Execute(cmds...); err != nil {
+		return fmt.Errorf("failed to delete chassis of name \"%s\" for node %s: error: %v",
```

Comment:
nit: consider error text to be `fmt.Errorf("failed to delete chassis %s for node %s": error %v", ...)`